### ISSUE TITLE
Foreach syntax and semantics too(?) complicated...

### ISSUE THREAD

**Issue by martinfabian** [Chalmers-Control-Automation-Mechatronics/Supremica#99]
I have been hesitant to file this as an issue, since I assume it is not really that but merely a fault with me behind the keyboard. Still, I find the Foreach syntax and semantics so confusing, that I doubt many others feel comfortable with it.

What I expect is to be able to write something like
```
FOR i IN 1..5
G[i]
```
and then this should give me five automata named G[1], G[2] etc, each with events like s[1], f[1] (for G[1]) and s[2], f[2] for G[2], etc. Then I want to be able to change the indices from 1..5, to say 1..9, and get nine automata instead with the events labeled appropriately. But achieving this is a big trial and error task for me, and it feels like I work quicker by copy-pasting five, or even nine, times and manually relabeling all events (and that is actually what always happens, I give up and do it all manually).

I look at the "dreitueren" model (Examples... > Module Examples > Central locking. dreitueren), and especially the "mer[TUER]" model, but there it seems like the events have to know beforehand, not only that they should be indexed, but also over which range, and this is very confusing. I do not know the range beforehand!

In short, can something be done to make this more user-friendly? I think that would improve significantly the usefulness of our tool.
